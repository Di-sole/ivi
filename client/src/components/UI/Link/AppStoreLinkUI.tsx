import styles from './AppStoreLinkUI.module.scss'

interface AppStoreUIProps {
    href: string;
    logo: string;
    linkTexts: string[];
}

function AppStoreLinkUI({ href, logo, linkTexts }: AppStoreUIProps) {
    return (
        <a href={href} className={styles['appstore-link-wrapper']} role='link-to-store'>
            <img src={logo} alt="App store logo" className={styles['appstore-link-logo']}/>
            <div className={styles['appstore-link-textblock']}>
                <div className={styles['appstore-link-preamble']}>{linkTexts[0]}</div>
                <div className={styles['appstore-link-caption']}>{linkTexts[1]}</div>
            </div>
        </a>
    )
}

export default AppStoreLinkUI;