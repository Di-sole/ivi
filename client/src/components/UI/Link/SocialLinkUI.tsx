import styles from './SocialLinkUI.module.scss'

interface SocialLinkUIProps {
    href: string;
    logo: string;
}

function SocialLinkUI({ href, logo }: SocialLinkUIProps) {
    return (
        <a href={href} target='blank' role='link-to-media'>
            <div className={styles['social-link']}>
                <img className={styles['social-logo']} src={logo} alt="Social media logo" />
            </div>
        </a>
    )
}

export default SocialLinkUI;