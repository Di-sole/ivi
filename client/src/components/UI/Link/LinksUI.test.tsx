import React from 'react';
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react';
import SocialLinkUI from './SocialLinkUI';
import AppStoreLinkUI from './AppStoreLinkUI';

describe('LinksUI SNAPSHOTS TESTS', () => {
    test('SocialLinkUI should not be changed', () => {
        render(
            <SocialLinkUI
                href='https://vk.com/iviru?crc=fa4448c13e06e69ba9e814e8743c7e2e'
                logo='https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_vkontakte.svg'
            />
        )
        const link = screen.getByRole('link-to-media');

        expect(link).toMatchSnapshot()
    })

    test('AppStoreLinkUI should not be changed', () => {
        render(
            <AppStoreLinkUI
                href='https://go.onelink.me/app/devicesiOS'
                logo='https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/appleLogo.svg'
                linkTexts={['Загрузить в', 'App Store']}
            />
        )
        const link = screen.getByRole('link-to-store');

        expect(link).toMatchSnapshot()
    })
})