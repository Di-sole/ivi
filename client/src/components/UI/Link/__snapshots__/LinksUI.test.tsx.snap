// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksUI SNAPSHOTS TESTS AppStoreLinkUI should not be changed 1`] = `
<a
  class="appstore-link-wrapper"
  href="https://go.onelink.me/app/devicesiOS"
  role="link-to-store"
>
  <img
    alt="App store logo"
    class="appstore-link-logo"
    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/appleLogo.svg"
  />
  <div
    class="appstore-link-textblock"
  >
    <div
      class="appstore-link-preamble"
    >
      Загрузить в
    </div>
    <div
      class="appstore-link-caption"
    >
      App Store
    </div>
  </div>
</a>
`;

exports[`LinksUI SNAPSHOTS TESTS SocialLinkUI should not be changed 1`] = `
<a
  href="https://vk.com/iviru?crc=fa4448c13e06e69ba9e814e8743c7e2e"
  role="link-to-media"
  target="blank"
>
  <div
    class="social-link"
  >
    <img
      alt="Social media logo"
      class="social-logo"
      src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_vkontakte.svg"
    />
  </div>
</a>
`;
